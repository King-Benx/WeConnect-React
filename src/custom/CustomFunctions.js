import {
    NotificationManager
} from 'react-notifications';
class CustomFunctions {
    static createNotifications(code, message) {
        // Generate notifications based off status codes
        switch (code) {
            case 200:
                return NotificationManager.success(message, 'Success');
            case 404:
            case 500:
                return NotificationManager.error(message);
            case 201:
                return NotificationManager.info(message);
            case 400:
                return NotificationManager.warning(message);
            case 401:
                return NotificationManager.error(message);
            default:
                return NotificationManager.error(message);
        }
    }

    static storeToken(token) {
        // Store a token and clear it after it expires
        localStorage.clear();
        localStorage.setItem('token', token)
    }

    static checkAuthentication() {
        // check whether token exists
        const stored_token = localStorage.getItem('token')
        if (stored_token) {
            return stored_token && stored_token.length > 10;
        } else {
            return false;
        }
    }

    static getToken() {
        // return token
        return localStorage.getItem('token')
    }

    static deleteToken() {
        // delete token from local storage
        localStorage.clear()
        localStorage.removeItem('token')
        return true
    }
}

export default CustomFunctions;