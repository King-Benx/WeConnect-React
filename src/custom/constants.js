// BASE URL OF WECONNECT-API
export const BASE_URL = "https://weconnect-api-database.herokuapp.com/";