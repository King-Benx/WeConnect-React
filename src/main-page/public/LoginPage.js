import React from 'react';
import {Link, Redirect} from 'react-router-dom'
import { Button } from 'react-bootstrap';
import superagent from 'superagent';
import { BASE_URL } from '../functional-resources/urls';
import CustomFunctions from '../functional-resources/CustomFunctions';
class LoginPage extends React.Component{
    constructor(){
        super();
        this.state = {email:"",password:""};
    }

    handleChange = (event)=>{
        this.setState({[event.target.id]: event.target.value});
    }

    handleReset= (event)=>{
        this.setState({email:"",password:""});
    }

    formSubmit(event){
        event.preventDefault();
        superagent
            .post(BASE_URL+'api/v1/auth/login')
            .send({email:this.state.email, password:this.state.password})
            .end((err,res)=>{
                if(err){CustomFunctions.createNotifications(err.status, res.body.message);}; 
                CustomFunctions.storeToken(res.body.message.token, 20)
                this.setState();
                this.props.history.push('/dashboard');
            });
    }

    render(){
        return ( 
            CustomFunctions.checkAuthentication()? <Redirect to={{pathname:'/dashboard'}}/>:
            <div className="row">
                <h1 className="text-center">
                    <em>WeConnect</em>
                </h1>
                <div id="login-panel" className="panel panel-default">
                    <div className="panel-heading">
                        <h3 className="panel-title">Login</h3>
                    </div>
                    <div className="panel-body">
                        <form method="POST" className="form" onSubmit={this.formSubmit.bind(this)}>
                            <div className="form-group">
                                <div className="input-group">
                                    <span className="input-group-addon" id="email-addon">Email</span>
                                    <input type="email" className="form-control" value={this.state.email} name="email" id="email" onChange={this.handleChange.bind(this)} placeholder="johndoe@mail.com" aria-describedby="email-addon" required="required"/>
                                </div>
                            </div>
                            <div className="form-group">
                                <div className="input-group">
                                    <span className="input-group-addon" id="password-addon">Password</span>
                                    <input type="password" className="form-control" value={this.state.password} name="password" id="password" onChange={this.handleChange.bind(this)} placeholder="Password" aria-describedby="password-addon" required="required"/>
                                </div>
                            </div>
                            <div className="form-group">
                                <Button type="reset" onClick={this.handleReset.bind(this)} className="pull-left"><i className="glyphicon glyphicon-refresh"></i> Reset</Button>
                                <Button type="submit" className="pull-right" bsStyle="success"><i className="glyphicon glyphicon-log-in"></i> Login</Button>
                                <div className="clearfix"></div>
                            </div>
                        </form>
                        <div className="text-center">
                            <Link to="/">Not a member? Register Today!</Link>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default LoginPage;