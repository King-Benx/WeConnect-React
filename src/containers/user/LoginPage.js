import React from "react";
import { Link } from "react-router-dom";
import { Button } from "react-bootstrap";
import axios from "axios";
import { BASE_URL } from "../../custom/constants";
import CustomFunctions from "../../custom/CustomFunctions";
class LoginPage extends React.Component {
  constructor() {
    super();
    this.state = { email: "", password: "" };
  }

  handleChange = event => {
    // handles change of state on input change
    this.setState({ [event.target.id]: event.target.value });
  };

  handleReset = event => {
    // handles reset of the login form
    this.setState({ email: "", password: "" });
  };

  formSubmit = event => {
    // handles form submit of login page
    event.preventDefault();
    axios
      .post(BASE_URL + "api/v1/auth/login", {
        email: this.state.email,
        password: this.state.password
      })
      .then(res => {
        CustomFunctions.storeToken(res.data.message.token);
        this.setState();
        this.props.history.push("/dashboard");
      })
      .catch(err => {
        CustomFunctions.createNotifications(
          err.status,
          err.response.data.message
        );
      });
  };

  render() {
    return (
      // JSX returned
      <div className="row">
        <h1 className="text-center">
          <em>WeConnect</em>
        </h1>
        <div id="login-panel" className="panel panel-default">
          <div className="panel-heading">
            <h3 className="panel-title">Login</h3>
          </div>
          <div className="panel-body">
            <form method="POST" className="form" onSubmit={this.formSubmit}>
              <div className="form-group">
                <input
                  type="email"
                  className="form-control"
                  value={this.state.email}
                  name="email"
                  id="email"
                  onChange={this.handleChange}
                  placeholder="Enter email eg. johndoe@mail.com"
                  aria-describedby="email-addon"
                  required="required"
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  value={this.state.password}
                  name="password"
                  id="password"
                  onChange={this.handleChange}
                  placeholder="Password"
                  aria-describedby="password-addon"
                  required="required"
                />
              </div>
              <div className="form-group">
                <Button
                  type="reset"
                  onClick={this.handleReset}
                  className="pull-left"
                >
                  <i className="glyphicon glyphicon-refresh" /> Reset
                </Button>
                <Button type="submit" className="pull-right" bsStyle="success">
                  <i className="glyphicon glyphicon-log-in" /> Login
                </Button>
                <div className="clearfix" />
              </div>
            </form>
            <div className="text-center">
              <Link to="/">Not a member? Register Today!</Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default LoginPage;
