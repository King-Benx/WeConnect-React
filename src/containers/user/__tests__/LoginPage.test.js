import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from "../LoginPage";
import { BASE_URL } from '../../../custom/constants';
import renderer from 'react-test-renderer';

describe('These are tests for the login form ', () =>{
    const login_route = BASE_URL+'api/v1/auth/login';
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }

    let mountedLoginPage;

    beforeEach(()=>{  
        mountedLoginPage= shallow(<LoginPage history={{push: ()=>{}}}/>);
    });

    it('handles input changed', () => {
        mountedLoginPage.instance().handleChange(event)
    })

    it('handles input reset', () => {
        mountedLoginPage.instance().handleReset(event)
    })

    // it('handles form submit', () => {
    //     mountedLoginPage.instance().formSubmit(event)
    // })
    
    it('renders without crashing', () => {
        shallow(<LoginPage />);
        const tree = renderer.create(<LoginPage />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has an h3', ()=>{
        const h3s = mountedLoginPage.find('h3')
        expect(h3s.length).toBe(1)
    })

    it('has an em', () => {
        const ems = mountedLoginPage.find('em')
        expect(ems.length).toBe(1)
    });

    it('has ah h1', () => {
        const his = mountedLoginPage.find('h1')
        expect(his.length).toBe(1)
    });

    it('renders the form', () => {
        const forms = mountedLoginPage.find('form')
        expect(forms.length).toBe(1)
    });

    it('renders form input', () => {
        const inputs = mountedLoginPage.find('input')
        expect(inputs.length).toBe(2)
    });

    it('has divs', () => {
        const divs = mountedLoginPage.find('div')
        expect(divs.length).toBe(9)
    });

    it('has links', () => {
        const links = mountedLoginPage.find('Link')
        expect(links.length).toBe(1)
    });

    it('has buttons', () => {
        const buttons = mountedLoginPage.find('Button')
        expect(buttons.length).toBe(2)
    });
});