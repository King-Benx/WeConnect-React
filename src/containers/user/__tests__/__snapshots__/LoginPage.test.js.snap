// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the login form  renders without crashing 1`] = `
<div
  className="row"
>
  <h1
    className="text-center"
  >
    <em>
      WeConnect
    </em>
  </h1>
  <div
    className="panel panel-default"
    id="login-panel"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Login
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <form
        className="form"
        method="POST"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            aria-describedby="email-addon"
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter email eg. johndoe@mail.com"
            required="required"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            aria-describedby="password-addon"
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required="required"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="pull-left btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="reset"
          >
            <i
              className="glyphicon glyphicon-refresh"
            />
             Reset
          </button>
          <button
            className="pull-right btn btn-success"
            disabled={false}
            type="submit"
          >
            <i
              className="glyphicon glyphicon-log-in"
            />
             Login
          </button>
          <div
            className="clearfix"
          />
        </div>
      </form>
      <div
        className="text-center"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Not a member? Register Today!
        </a>
      </div>
    </div>
  </div>
</div>
`;
